feat(embed-hub): checklist endpoint

npretto opened the pull request:
Closes [EMB-772: move the settings for "create test/prod embed" to internal (if possible) and make the endpoint read them](https://linear.app/metabase/issue/EMB-772/move-the-settings-for-create-testprod-embed-to-internal-if-possible)
Closes [EMB-770: Create an endpoint for embedding hub checklist completion data](https://linear.app/metabase/issue/EMB-770/create-an-endpoint-for-embedding-hub-checklist-completion-data)
Closes [EMB-768: Move the "sandbox" check to the embedding hub endpoint](https://linear.app/metabase/issue/EMB-768/move-the-sandbox-check-to-the-embedding-hub-endpoint)
Closes [EMB-742: Auto-mark "x-ray a table" step as done when a dashboard is present](https://linear.app/metabase/issue/EMB-742/auto-mark-x-ray-a-table-step-as-done-when-a-dashboard-is-present)
Closes [EMB-740: Auto-mark "connect a database" step as done when databases are added](https://linear.app/metabase/issue/EMB-740/auto-mark-connect-a-database-step-as-done-when-databases-are-added)

This pr introduces a new endpoint that will return if the steps of the embedding checklist have been done or not:
<img width="847" height="593" alt="image" src="https://github.com/user-attachments/assets/0ef7b5f6-6070-433e-a236-e3006034cf95" />

I created the endpoint using LLMs trying to check what they did but my clojure knowledge is limited and I don't know how the backend code is structured so I focused on the "making it work part", leaving the "making it pretty" and the tests for a real BE developer.

npretto committed:
feat(embed-hub): EMB-770 and EMB-742, add checklist endpoint with
'create-dashboard'

npretto committed:
feat(embed-hub): EMB-740 auto mark "Add data" as done

npretto committed:
Revert "mark sandboxed step as done (#62758)"

This reverts commit be775aceed7f0338cc46749f5accb1e12009bb92.

npretto committed:
feat(embed-hub): EMB-768 move sandbox check to the BE

npretto committed:
EMB-772: move all "done" checks on the be, move endpoints to ee

npretto committed:
cleanup

fmt

configure enterprise module to make linting pass

test: activate license to make e2e tests pass again

remove fe unit test now that all the logic is on the be

add doc to the endpoint to maybe the tests will pass?

whitespace?

fixes

idk anymore how to make all the linters be happy

last try

refactor has-configured-sso to a function

revert  visibility to see if tests pass

sort config.edn

heypoom commented on frontend/src/metabase/embedding/embedding-hub/hooks/use-completed-embedding-hub-steps.ts:
```diff
@@ -1,67 +1,35 @@
+} => {
+  const { data: embeddingHubChecklist, isLoading } =
+    useGetEmbeddingHubChecklistQuery(undefined, {
+      refetchOnMountOrArgChange: true,
```
I suppose this re-fetches the checklist every time the page visited, which is good so we can keep the checklist updated after key actions 👍🏻

heypoom commented on enterprise/backend/src/metabase_enterprise/api/routes.clj:
```diff
@@ -100,6 +102,7 @@
    "/database-routing"             (premium-handler metabase-enterprise.database-routing.api/routes :database-routing)
    "/document"                     (premium-handler metabase-enterprise.documents.api/routes :documents)
    "/email"                        (premium-handler metabase-enterprise.email.api/routes :cloud-custom-smtp)
+   "/embedding-hub"                (premium-handler metabase-enterprise.embedding-hub.api/routes :embedding)
```
This makes sense, since we currently gate the embedding hub under `:embedding` and is likely to stay that way.

heypoom reviewed (approved):
Frontend-wise and E2E-wise this looks good, will leave the backend details for BE folks to check 🙏🏻

npretto committed:
tests: remove cy.wait from embed hub tests (#63066)

dpsutton committed:
Bit of cleanup

- remove warn-on-reflection; there is no interop
- clean up a bit of duplication in sql clause generation
- remove unncessary try/catch around t2 access
- remove a few boolean calls on boolean functions
